PS5Controller/PS5Controller/ViewController.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  ViewController.swift
+//  PS5Controller
+//
+//  Created by Alperen Aysel on 6.05.2021.
+//
+
+import UIKit
+
+final class ViewController: UIViewController {
+
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var controllerTabView: UIView!
+    @IBOutlet private weak var devicesTabView: UIView!
+    @IBOutlet private weak var mouseTabView: UIView!
+    
+    private let products: [Product] = [.init(image: UIImage(named: "whitePSController")!),
+                               .init(image: UIImage(named: "blackPSController")!),
+                               .init(image: UIImage(named: "whitePSController")!)]
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        collectionView.dataSource = self
+        collectionView.register(ControllerCell.nib(), forCellWithReuseIdentifier: "ControllerCell")
```

Comment:
bunu her cell için yapıyosun, extension'a taşıyıp sadece collectionView.register(type: ControllerCollectionViewCell.self) şeklinde nası yaparsın? Hep her cell'in nibName'i class adı, identifier'ı da class adı olsun

Reply:
```
extension UICollectionViewCell {
    static func nib(withName nibName: String) -> UINib {
        return UINib(nibName: nibName, bundle: nil)
    }
}
```
Şöyle bir extension oluşturup

```
collectionView.register(ControllerCell.nib(withName: "ControllerCell"), forCellWithReuseIdentifier: "ControllerCell")
```

Olarak  kullanabilirdim her cell'e tekrardan nib fonksiyonu yazmazdım böylece ama senin dediğin gibi nasıl yapıcağımı çözemedim :/ Kafa patlatıyorum ama .self diyince nasıl register için gerekli değerlerin hepsini dönücem bulamadım